### PR TITLE
Update interface for DIRECT 7.0.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

File: interoplib.h [hunk 1]
```diff
@@ -10,133 +10,135 @@ extern "C" {
 #include <stdio.h>
 #include <stdint.h>
 #include <string.h>
+#include <stdbool.h>
 
 /*********************************************************************/
 
-typedef uint8_t                                 uint8;
```
Comment (sergio-nsk) [inline interoplib.h:16]: The header guard has not been updated.

**Comment by nmoinvaz** [comment]
This PR is also ready.

**Review by sergio-nsk — CHANGES_REQUESTED** [review]

File: interoplib.h (hunk 1 above)
Comment (nmoinvaz) [reply interoplib.h:16]: Thanks I have fixed it.

**Comment by sergio-nsk** [comment]
Too much changes. Will take look at them a bit later.

**Comment by nmoinvaz** [comment]
Ok, most of it is just copying code from core. Also we have build and lint checks.

**Comment by sergio-nsk** [comment]
Are you sure there are no errors with the return types `bool` where `int32_t` are expected?

**Review by sergio-nsk — APPROVED** [review]

**Comment by nmoinvaz** [comment]
I don't think so. I copied from core sources so..

**Pull request merged**